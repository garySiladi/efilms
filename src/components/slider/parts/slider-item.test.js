// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import SliderItem from './slider-item';

const dummyData = [
  {
    title: 'Episode 6 Title',
    type: 'Episode',
    id: 6,
    subtitle: 'subtitle',
    // $FlowFixMe
    thumbnail: {
      thumb: {
        url: 'url',
      },
    },
    className: 'slider-item',
    isSelected: false,
    isBeforeSelected: false,
    isSelectedSeries: false,
    isEpisodeExpanded: false,
  },
  {
    title: 'Series 1 Title',
    type: 'Series',
    id: 9,
    episode_count: 10,
    // $FlowFixMe
    thumbnail: {
      thumb: {
        url: 'url',
      },
    },
    className: 'slider-item',
    isSelected: false,
    isBeforeSelected: true,
    isSelectedSeries: true,
    isEpisodeExpanded: false,
  },
  {
    title: 'Episode 19 Title',
    type: 'Episode',
    id: 19,
    subtitle: 'subtitle',
    // $FlowFixMe
    thumbnail: {},
    className: 'slider-item',
    isSelected: true,
    isBeforeSelected: false,
    isSelectedSeries: true,
    isEpisodeExpanded: false,
  },
  {
    title: 'Episode 19 Title',
    type: 'Episode',
    id: 19,
    subtitle: 'subtitle',
    // $FlowFixMe
    thumbnail: {},
    className: 'slider-item',
    isSelected: true,
    isBeforeSelected: false,
    isSelectedSeries: false,
    isEpisodeExpanded: true,
  },
];

const fullWidthDummyData = [
  {
    title: 'Episode 1 Title',
    type: 'Episode',
    id: 1,
    subtitle: 'subtitle',
    // $FlowFixMe
    thumbnail: {
      full_width: {
        url: 'url',
      },
    },
    className: 'slider-item',
    isSelected: false,
    isBeforeSelected: false,
    isSelectedSeries: false,
    isEpisodeExpanded: false,
  },
  {
    title: 'Episode 2 Title',
    type: 'Episode',
    id: 2,
    subtitle: 'subtitle',
    // $FlowFixMe
    thumbnail: {},
    className: 'slider-item',
    isSelected: false,
    isBeforeSelected: false,
    isSelectedSeries: false,
    isEpisodeExpanded: false,
  },
];

test('SliderItem renders correctly', () => {
  dummyData.forEach((data) => {
    const tree: string = renderer.create(<SliderItem {...data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

test('SliderItem renders correctly when full-width', () => {
  fullWidthDummyData.forEach((data) => {
    const tree: string = renderer.create(<SliderItem {...data} isFullWidth />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
