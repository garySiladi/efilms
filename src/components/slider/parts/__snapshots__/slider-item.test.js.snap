// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderItem renders correctly 1`] = `
<div
  className="slider-item"
>
  <div
    className="image image--placeholder"
  >
    <div
      className="image__holder"
    >
      <img
        alt="Episode 6 Title"
        className="slider-item__image"
        onLoad={[Function]}
        src="url"
      />
    </div>
  </div>
  <div
    className="slider-item__container"
  >
    <div
      className="slider-item__title"
    >
      Episode 6 Title
    </div>
    <div
      className="slider-item__subtitle"
    >
      subtitle
    </div>
  </div>
  <div
    className="slider-item__chevron"
  />
</div>
`;

exports[`SliderItem renders correctly 2`] = `
<div
  className="slider-item slider-item--series-item"
>
  <div
    className="image image--placeholder"
  >
    <div
      className="image__holder"
    >
      <img
        alt="Series 1 Title"
        className="slider-item__image"
        onLoad={[Function]}
        src="url"
      />
    </div>
  </div>
  <div
    className="slider-item__container"
  >
    <div
      className="slider-item__title"
    >
      Series 1 Title
    </div>
    <div
      className="slider-item__subtitle"
    >
      10 episodes
    </div>
  </div>
  <div
    className="slider-item__chevron"
  />
</div>
`;

exports[`SliderItem renders correctly 3`] = `
<div
  className="slider-item slider-item--selected"
>
  <div
    className="image image--placeholder"
  >
    <div
      className="image__holder"
    >
      <img
        alt="Episode 19 Title"
        className="slider-item__image"
        onLoad={[Function]}
        src=""
      />
    </div>
  </div>
  <div
    className="slider-item__container"
  >
    <div
      className="slider-item__title"
    >
      Episode 19 Title
    </div>
    <div
      className="slider-item__subtitle"
    >
      subtitle
    </div>
  </div>
  <div
    className="slider-item__chevron"
  />
</div>
`;

exports[`SliderItem renders correctly 4`] = `
<div
  className="slider-item"
>
  <div
    className="image image--placeholder"
  >
    <div
      className="image__holder"
    >
      <img
        alt="Episode 19 Title"
        className="slider-item__image"
        onLoad={[Function]}
        src=""
      />
    </div>
  </div>
  <div
    className="slider-item__container"
  >
    <div
      className="slider-item__title"
    >
      Episode 19 Title
    </div>
    <div
      className="slider-item__subtitle"
    >
      subtitle
    </div>
  </div>
  <div
    className="slider-item__chevron slider-item__chevron--visible"
  />
</div>
`;

exports[`SliderItem renders correctly when full-width 1`] = `
<div
  className="slider-item slider-item--full-width"
>
  <div
    className="image image--placeholder"
  >
    <div
      className="image__holder"
    >
      <img
        alt="Episode 1 Title"
        className="slider-item__image"
        onLoad={[Function]}
        src="url"
      />
    </div>
  </div>
  <div
    className="slider-item__chevron slider-item__chevron--small"
  />
</div>
`;

exports[`SliderItem renders correctly when full-width 2`] = `
<div
  className="slider-item slider-item--full-width"
>
  <div
    className="image image--placeholder"
  >
    <div
      className="image__holder"
    >
      <img
        alt="Episode 2 Title"
        className="slider-item__image"
        onLoad={[Function]}
        src=""
      />
    </div>
  </div>
  <div
    className="slider-item__chevron slider-item__chevron--small"
  />
</div>
`;
